### PR TITLE
feature: migrate to react 16

### LINKED ISSUE #135: React v16.0 & Popover issue
Hi, I just upgraded my app to [React v16.0](https://facebook.github.io/react/blog/), everything is working fine but the <Popover> component do not render anymore, here is the error reported in the console:
```
index.js:199 Uncaught TypeError: Failed to execute 'getComputedStyle' on 'Window': 
parameter 1 is not of type 'Element'.
    at Object.resolvePopoverLayout (index.js:199)
    at Object.trackPopover (index.js:429)
    at Object.enter (index.js:328)
    at Object.componentDidUpdate (index.js:153)
    at Object.chainedFunction [as componentDidUpdate] (factory.js:615)
    at commitLifeCycles (react-dom.development.js:11517)
    at commitAllLifeCycles (react-dom.development.js:12294)
    at HTMLUnknownElement.callCallback (react-dom.development.js:1299)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:1338)
    at invokeGuardedCallback (react-dom.development.js:1195)
    at commitAllWork (react-dom.development.js:12415)
    at workLoop (react-dom.development.js:12687)
    at HTMLUnknownElement.callCallback (react-dom.development.js:1299)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:1338)
    at invokeGuardedCallback (react-dom.development.js:1195)
    at performWork (react-dom.development.js:12800)
    at batchedUpdates (react-dom.development.js:13244)
    at performFiberBatchedUpdates (react-dom.development.js:1646)
    at stackBatchedUpdates (react-dom.development.js:1637)
    at batchedUpdates (react-dom.development.js:1651)
    at Object.batchedUpdatesWithControlledComponents [as batchedUpdates] (react-dom.development.js:1664)
    at dispatchEvent (react-dom.development.js:1874)
```

Thanks for help

### PR DESCRIPTION
**Pull request by jasonkuhrt**
Will close #135.

- [x] react 16
- [x] class syntax
- [x] no custom portal
- [x] remove unused test deps

### CODE REVIEW

**Comment by jasonkuhrt** [comment]
@knowbody here's my start. It seems to be working so far. Going to tweak a few more things and see if I can break anything.

**Comment by Kerumen** [comment]
Working? The first layout glitch is still there. Even worse. (NB: I clicked once, for the first toggle)

kapture 2017-10-03 at 10 14 14

Latest Chrome.

**Comment by jebarjonet** [comment]
On my computer (and latest Chrome) it works well, that is a strange behavior

**Comment by Kerumen** [comment]
Ok I managed to narrow the bug. It appears only when you resize the "Action Logger" until the component window is exactly `554px` height (displayed in the bottom right corner). It seems to be an edge case where the positioning system can't find the spot between `above` and `right`.

**Comment by jasonkuhrt** [comment]
@Kerumen Hey, yeah I've seen that before and I think there was an issue at one time about it. Maybe its come back... It looks like an [instance of this edge case](https://blog.jasonkuhrt.com/essays/forto/#app-b). Integrating Forto #95 should make this issue go away for good though. Anyways not sure this is an issue particular to this PR. I'll try recreating on master.

**Comment by jasonkuhrt** [comment]
@Kerumen I'm not able to reproduce the issue at `554px` height locally or at https://littlebits.github.io/react-popover. But I don't doubt what you saw. I don't see an evidence that this PR is the culprit though.

**Comment by Kerumen** [comment]
@jasonkuhrt Yes the particular `554px` edge case is weird but it won't happen on a real app.
And you're right, I don't guess it's related to this PR.

File: package.json [hunk 1]
```diff
@@ -36,26 +35,18 @@
...
-    "cssnext-loader": "^1.0.1",
     "eslint": "^3.17.1",
     "eslint-config-littlebits": "^0.6.2",
-    "jest": "^19.0.2",
-    "jest-cli": "^19.0.2",
     "prettier": "^1.7.3",
     "ramda": "^0.18.0",
-    "react": "^15.4.2",
-    "react-dom": "^15.4.2",
+    "react": "16",
```
Comment (knowbody) [inline package.json]: probably would be better to not use exact v16 and instead use `^16.0.0`

File: lib/index.js [hunk 2]
```diff
@@ -74,33 +71,39 @@ const Popover = createReactClass({
...
+    isOpen: false,
+    onOuterAction: Utils.noop,
+    enterExitTransitionDurationMs: 500,
+    children: null,
+    refreshIntervalMs: 200,
+    appendTarget: isServer ? null : document.body
+  }
+  constructor (props) {
+    super(props)
+    this.checkTargetReposition = this.checkTargetReposition.bind(this)
```
Comment (knowbody) [inline lib/index.js]: if you use arrow functions then you wouldn't need to bind functions in the constructor:

```js
checkTargetReposition = () => {
  // ...
}
```

**Comment by knowbody** [comment]
thank you for working on this @jasonkuhrt!

File: package.json (hunk 1 above)
Comment (jasonkuhrt) [reply package.json]: Ah yeah, not used to `yarn upgrade@...` yet. Shouldn't we make react deps peer deps too?

Comment (jasonkuhrt) [reply package.json]: Yeah https://github.com/reactjs/react-modal/blob/master/package.json#L63-L66

Comment (jasonkuhrt) [reply package.json]: But meh this is still an issue https://github.com/yarnpkg/yarn/issues/2479 so whatever will leave peer deps definition for now.

**Comment by jasonkuhrt** [comment]
Thanks everyone for your feedback!

**Pull request merged**